### ISSUE TITLE
golint doesn't check xtest files

### ISSUE THREAD

**Issue by dominikh** [golang/lint#200]
Consider the following shell session:

```
dominikh-pc ~ $ GOROOT=~/go golint ~/go/src/net/http/serve_test.go        
/home/dominikh/go/src/net/http/serve_test.go:1070:6: don't use underscores in Go names; func testAutomaticHTTP2_ListenAndServe should be testAutomaticHTTP2ListenAndServe
dominikh-pc ~ $ GOROOT=~/go golint net/http | grep serve_test     
dominikh-pc 1? ~ $ 
$ GOROOT=~/go golint net/http | grep _test
/home/dominikh/go/src/net/http/export_test.go:26:2: var ExportHttp2ConfigureTransport should be ExportHTTP2ConfigureTransport
/home/dominikh/go/src/net/http/export_test.go:27:2: var ExportHttp2ConfigureServer should be ExportHTTP2ConfigureServer
/home/dominikh/go/src/net/http/readrequest_test.go:29:24: should drop = nil from declaration of var noTrailer; it is the zero value
```

When checking the net/http package, there are no linter warnings for the serve_test.go file. When checking that file individually, there are warnings. The third invocation is there to prove that it's looking in the right directory and that it does check tests, so that rules out two possible causes.

**Comment by dominikh** [comment]
The issue is that golint only checks the tests in package http, not in http_test.

**Comment by dsymonds** [comment]
Yeah, see golint.go:124:

```go
// TODO(dsymonds): Do foo_test too (pkg.XTestGoFiles)
```

**Comment by dominikh** [comment]
@dsymonds are you open to switching to `go/loader`, or would you prefer to keep the current code and fix it there?

**Comment by dominikh** [comment]
Disregard that, switching to go/loader would probably be a rather intrusive change. I'll see about fixing it in the current code.

**Comment by dsymonds** [comment]
#153 has been open for ages to do that, but I keep putting it off because it's such a non-trivial change.

External tests are relatively rare. I'm not sure it's worth the effort unless it's a relatively small change. They may also have different concerns about style.

**Comment by dominikh** [comment]
> I'm not sure it's worth the effort unless it's a relatively small change

Yeah, I discovered it in my golint fork, where it does matter, and figured I'd report it upstream. If it's not a high priority for you, I'll just fix it in my fork for now (via go/loader, most likely). At a quick glance I don't think that it's a small change one way or another, the xtest handling is a bit intricate if done manually.

**Comment by zimmski** [comment]
I would be happy to finish up #153. We are using the PR in production for a very long time now but I haven't got the time to test it with Go 1.6 since we are still on 1.4.

**Comment by akimd** [comment]
Still having the problem today.

**Comment by mvdan** [comment]
Thank you for submitting this issue! As per https://github.com/golang/go/issues/38968, we are freezing and deprecating golint. There's no drop-in replacement to golint per se, but you should find that [Staticcheck](https://staticcheck.io/) works well in encouraging good Go code, much like golint did in the past, since it also includes style checks. There's always `gofmt` and `go vet` too, of course.

If you would like to contribute further, I'd encourage you to engage [Staticcheck's issue tracker](https://github.com/dominikh/go-tools/issues) or look at [vet's open issues](https://github.com/golang/go/issues?q=is%3Aissue+is%3Aopen+cmd%2Fvet+in%3Atitle), as they are both actively maintained. If you have an idea that doesn't fit into either of those tools, you could look at [other Go linters](https://github.com/golangci/awesome-go-linters), or write your own - these days it's fairly straightforward with [go/analysis](https://pkg.go.dev/golang.org/x/tools/go/analysis).

To help avoid confusion, I'm closing all issues before we freeze the repository. If you have any feedback, you can leave a comment on the proposal thread where it was decided to deprecate golint - though note that the proposal has been accepted for nearly a year. Thanks!
